packages/core/src/crypto/jose/jwk/transform.ts

```diff
@@ -47,3 +47,17 @@ export function getJwkClassFromJwaSignatureAlgorithm(alg: JwaSignatureAlgorithm
 export function getJwkClassFromKeyType(keyType: KeyType) {
   return JwkClasses.find((jwkClass) => jwkClass.keyType === keyType)
 }
+
+/**
```

Comment:
Key types can support multiple JWAs. For that reason I choose to only support key -> jwk, and then on a JWK class you can see the supported algorithms.

This is going to be an issue down the line

Reply:
Yeah that is true. I would like to create the header for the user as we can then just accept a VC, which would be the payload, and nothing else. Do you have a suggestion on how we can create a header, which IIRC only contains the `alg` and `typ` as required?

Reply:
Yeah I think we can do that, but that we need to pick the default JWA differently than is done with this method.

See this code: https://github.com/hyperledger/aries-framework-javascript/blob/b2ba7c7197139e780cbb95eed77dc0a2ad3b3210/packages/core/src/crypto/JwsService.ts#L28-L36

From keyType we can go to JWK and from a JWK we can then pick the first supported signature algorithm. But we should allow the user to specify the alg, so they can override it.

You can get the JWK class using this method: https://github.com/hyperledger/aries-framework-javascript/blob/b2ba7c7197139e780cbb95eed77dc0a2ad3b3210/packages/core/src/crypto/jose/jwk/transform.ts

And then statically extract the signature algorithms from the class: https://github.com/hyperledger/aries-framework-javascript/blob/b2ba7c7197139e780cbb95eed77dc0a2ad3b3210/packages/core/src/crypto/jose/jwk/Ed25519Jwk.ts#L15